src/answers-headless.ts

```diff
@@ -320,7 +320,19 @@ export default class AnswersHeadless {
       context,
       referrerPageUrl,
       restrictVerticals
-    });
+    };
+
+    let response: UniversalSearchResponse;
+    try {
+      response = await this.core.universalSearch(request);
+    } catch (e) {
+      const latestResponseId = this.httpManager.getLatestResponseId('universalQuery');
```

Comment:
what do you think of adding a helper to httpManager to make this logic a little simpler? For example, we could have `isMostRecentUniversalRequest` which takes in a requestId and returns a boolean. I'm also considering having a `processUniversalRequestId` which returns whether or not it's the most recent id as a boolean, and if so, it automatically sets the responseId. I think either of those might help make this logic a tad more clear and concise. Such a change could be used elsewhere in headless so I'm okay if we make another item and consider it out of scope of this PR

Reply:
sure! I think the `processUniversalRequestId` approach sounds good, but probably make it a generic `processRequestId` that takes in a request type (e.g. 'universalQuery') as well. I can update